pinot-core/src/main/java/com/linkedin/pinot/core/query/selection/SelectionOperatorService.java

```diff
@@ -15,369 +15,153 @@
+   * Helper method to get the {@link Comparator} for selection rows.
    *
-   * @param reducedResults
-   * @param dataSchema
-   * @param offset
-   * @return
-   * @throws Exception
+   * @return {@link Comparator} for selection rows.
    */
-  public SelectionResults renderSelectionResults(Collection<Serializable[]> reducedResults, DataSchema dataSchema,
-      int offset)
-      throws Exception {
-    final LinkedList<Serializable[]> rows = new LinkedList<Serializable[]>();
-
-    if (reducedResults instanceof PriorityQueue<?>) {
-      PriorityQueue<Serializable[]> queue = (PriorityQueue<Serializable[]>) reducedResults;
-      while (reducedResults.size() > offset) {
-        rows.addFirst(SelectionOperatorUtils.getFormattedRow(queue.poll(), _selectionColumns, dataSchema));
-      }
-    } else if (reducedResults instanceof ArrayList<?>) {
-      List<Serializable[]> list = (List<Serializable[]>) reducedResults;
-      for (int i = offset; i < list.size(); i++) {
-        rows.add(SelectionOperatorUtils.getFormattedRow(list.get(i), _selectionColumns, dataSchema));
-      }
-    } else {
-      throw new UnsupportedDataTypeException(
-          "type of results Expected: (PriorityQueue| ArrayList)) actual:" + reducedResults.getClass());
-    }
-
-    return new SelectionResults(_selectionColumns, rows);
-  }
-
-  private JSONArray getSelectionColumnsJsonArrayFromDataSchema(DataSchema dataSchema) {
-    final JSONArray jsonArray = new JSONArray();
-    for (String column: _selectionColumns) {
-      jsonArray.put(column);
-    }
-    return jsonArray;
-  }
-
-  private List<String> getSelectionColumnsFromDataSchema(DataSchema dataSchema) {
-    List<String> columns = new ArrayList<String>();
-
-    for (int i = 0; i < dataSchema.size(); i++) {
-      if (_selectionColumns.contains(dataSchema.getColumnName(i))) {
-        columns.add(dataSchema.getColumnName(i));
-      }
-    }
-    return columns;
-  }
-
-  public Collection<Serializable[]> getRowEventsSet() {
-    return _rowEventsSet;
-  }
-
-  public DataSchema getDataSchema() {
-    return _dataSchema;
-  }
-
-  public long getNumDocsScanned() {
-    return _numDocsScanned;
-  }
-
-  private Comparator<Serializable[]> getComparator(final List<SelectionSort> sortSequence,
-      final DataSchema dataSchema) {
+  private Comparator<Serializable[]> getComparator() {
     return new Comparator<Serializable[]>() {
       @Override
       public int compare(Serializable[] o1, Serializable[] o2) {
-        for (int i = 0; i < sortSequence.size(); ++i) {
+        int numSortColumns = _sortSequence.size();
+        for (int i = 0; i < numSortColumns; i++) {
           int ret = 0;
-          switch (dataSchema.getColumnType(i)) {
+          // Only compare single-value columns.
+          switch (_dataSchema.getColumnType(i)) {
             case INT:
-              if (!sortSequence.get(i).isIsAsc()) {
+              if (!_sortSequence.get(i).isIsAsc()) {
```

Comment:
Because its a private helper method, so no need to pass in members as arguments.